### PR TITLE
flecsi: add mpi interoperability

### PR DESCRIPTION
**Pull request by junghans**
Part of laristra/flecsi#235.

### CODE REVIEW

File: var/spack/repos/builtin/packages/flecsi/package.py [hunk 1]
```diff
@@ -61,4 +64,10 @@ def build_type(self):
             return 'Release'
 
     def cmake_args(self):
-        return ['-DENABLE_UNIT_TESTS=ON']
+        options = ['-DENABLE_UNIT_TESTS=ON']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flecsi/package.py]: Does this flag cause the unit tests to run? If so, you should make it optional:
```python
options = []

if self.run_tests:
    options.append('-DENABLE_UNIT_TESTS=ON')
```
Then the tests will be run only when a user explicitly asks for them:
```
$ spack install --run-tests
```

Comment (junghans) [inline var/spack/repos/builtin/packages/flecsi/package.py]: No, it won't run the tests, it just builds them. I just like to build an executable to see if all symbols get resolved.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flecsi/package.py]: I'm glad I'm not the only one checking these things 😄

File: var/spack/repos/builtin/packages/flecsi/package.py [hunk 2]
```diff
@@ -39,19 +39,22 @@ class Flecsi(CMakePackage):
     homepage = "http://flecsi.lanl.gov/"
     url      = "https://github.com/laristra/flecsi/tarball/v1.0"
 
-    version('develop', git='https://github.com/laristra/flecsi', branch='master', submodules=True)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/flecsi/package.py]: you don't have to do this, `version()` will not be checked for line width.

File: var/spack/repos/builtin/packages/flecsi/package.py [hunk 3]
```diff
@@ -61,4 +64,10 @@ def build_type(self):
             return 'Release'
 
     def cmake_args(self):
-        return ['-DENABLE_UNIT_TESTS=ON']
+        options = ['-DENABLE_UNIT_TESTS=ON']
+
+        if '+mpi' in self.spec:
+            options.extend(['-DENABLE_MPI=ON',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/flecsi/package.py]: i think we normally format it as
```
options.extend([
    blabla
])
```

**Comment by junghans** [comment]
@davydden : Happy now?

**Pull request merged**